source/postprocess/visualization/compositional_vector.cc

```diff
@@ -0,0 +1,202 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/postprocess/visualization/compositional_vector.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      CompositionalVector<dim>::
+      CompositionalVector ()
+        :
+        DataPostprocessor<dim> ()
+      {}
+
+
+      template <int dim>
+      std::vector<std::string>
+      CompositionalVector<dim>::
+      get_names () const
+      {
+        std::vector<std::string> solution_names;
+        for (unsigned int i=0; i<vector_names.size(); ++i)
+          for (unsigned int j=0; j<dim; ++j)
+            solution_names.push_back(vector_names[i]);
```

Comment:
You're creating a vector of strings by pushing back the elements of a vector of strings. You could just return the original vector of strings here :-)